server/fleet/mdm.go

```diff
@@ -392,3 +400,105 @@ func NewMDMConfigProfilePayloadFromApple(cp *MDMAppleConfigProfile) *MDMConfigPr
 		UpdatedAt:   cp.UpdatedAt,
 	}
 }
+
+// MDMProfileSpec represents the spec used to define configuration
+// profiles via yaml files.
+type MDMProfileSpec struct {
+	Path   string   `json:"path,omitempty"`
+	Labels []string `json:"labels,omitempty"`
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface to add backwards
+// compatibility to previous ways to define profile specs.
+func (p *MDMProfileSpec) UnmarshalJSON(data []byte) error {
+	if len(data) == 0 {
+		return nil
+	}
+
+	// use an alias type to avoid recursively calling this function
+	// forever.
+	type Alias MDMProfileSpec
+	aliasData := struct {
+		*Alias
+	}{
+		Alias: (*Alias)(p),
+	}
+	newFormatErr := json.Unmarshal(data, &aliasData)
+	if newFormatErr == nil {
+		return nil
+	}
+
+	var backwardsCompat string
+	oldFormatErr := json.Unmarshal(data, &backwardsCompat)
+	if newFormatErr != nil && oldFormatErr != nil {
+		return fmt.Errorf("unmarshal profile spec. Error using new format: %w. Error using old format: %w", newFormatErr, oldFormatErr)
+	}
+	p.Path = backwardsCompat
```

Comment:
Nice! Just to save on the double unmarshal, we can just look at the first byte see if it's "{" or '"'. I'll update it if there's time (not critical).